mobile/library/cc/stream_callbacks.cc

```diff
@@ -110,7 +124,7 @@ envoy_http_callbacks StreamCallbacks::asEnvoyHttpCallbacks() {
       &c_on_complete,
       &c_on_cancel,
       &c_on_send_window_available,
-      new StreamCallbacksSharedPtr(shared_from_this()),
+      stream_callbacks_wrapper,
```

Comment:
Hm... After reading through the changes in this PR, I kinda think I like the existing code better. But I suspect I should feel bad about that. Maybe let's chat on Monday. But my thinking is... The thing that we need to store in the `void*` (directly or indirectly) is a shared_ptr (*cough* stupid Envoy type aliasing *cough*) so that we can ensure that the callbacks are kept alive. This `StreamCallbacksWrapper` is, if I understand correctly, a struct that exists simply so that we can wrap the `shared_ptr` in a heap allocated struct in order to avoid it going out of scope. I kinda feel like this is simply less readable than having a single `new StreamCallbacksSharedPtr(shared_from_this())` which we could decorate which some hefty comments.

But lets chat offline. I'm probably not thinking about this clearly.

Reply:
As discussed offline, Ryan is right, a clarifying comment is best while keeping the code the same, so I reverted the change and instead added a (long) clarifying comment.